lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3388,26 +3388,23 @@ namespace {
                     ->getTopLevelModule()
                     ->getFullModuleName() == n;
         };
-        if ((decl->getOwningModule() &&
-             importer::isCxxStdModule(decl->getOwningModule()))) {
+        clang::Module *owningModule = decl->getOwningModule();
+        if ((owningModule && importer::isCxxStdModule(owningModule))) {
           if (isAlternativeCStdlibFunctionFromTextualHeader(decl)) {
             return nullptr;
           }
-          auto filename =
-              Impl.getClangPreprocessor().getSourceManager().getFilename(
-                  decl->getLocation());
-          if (filename.endswith("cmath") || filename.endswith("math.h") ||
-              filename.endswith("stdlib.h") || filename.endswith("cstdlib")) {
-            return nullptr;
+
+          auto &sourceManager = Impl.getClangPreprocessor().getSourceManager();
+          if (const auto file = sourceManager.getFileEntryForID(
+                  sourceManager.getFileID(decl->getLocation()))) {
+            auto filename = file->getName();
+            if ((file->getDir() == owningModule->Directory) &&
+                (filename.endswith("cmath") || filename.endswith("math.h") ||
+                 filename.endswith("stdlib.h") || filename.endswith("cstdlib"))) {
```

Comment:
This matches the existing behavior, but I don't understand why we're rejecting everything in these headers. It feels like we should only reject them if they are one of the likely conflicting symbols.

Reply:
We should probably annotate them as unavailable using `std.apinotes`

Reply:
Yeah I don't know why we're dumping the whole files instead of adding the problem declarations to isAlternativeCStdlibFunctionFromTextualHeader, or why the C++ declarations are even a problem in the first place. It seems like we'd _want_ the C++ declarations generally.

But I'll leave that for a future change 😄